lib/location/location_core.c

```diff
@@ -383,6 +383,13 @@ static int location_core_location_get_pos(void)
 
 	err = location_method_api_get(requested_method)->location_get(&loc_req_info);
 
+	if (IS_ENABLED(CONFIG_LOCATION_DATA_DETAILS) && err == 0) {
```

Comment:
There's now two `if`'s checking for the same error. Those could replace with one check and in case of an error it could be returned right after `location_get()`. The rest of the function would be executed only in success case.

Reply:
Hope it went right...